servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -568,4 +570,16 @@ public void sessionHeartbeat(long sessionId, List<Long> metrics) {
   public void updatePinList(Set<Long> pinnedInodes) {
     mBlockStore.updatePinnedInodes(pinnedInodes);
   }
+
+  /**
+   * Gets the file information.
+   *
+   * @param fileId the file id
+   * @return the file info
+   * @throws FileDoesNotExistException if the file does not exist
+   * @throws IOException if an I/O error occurs
+   */
+  public FileInfo getFileInfo(long fileId) throws FileDoesNotExistException, IOException {
```

Comment:
Good point. I had a refactoring to do for it. Basically I found it weird that `BlockDataManager` contains `mFileSystemMasterClient`. Basically all the workers have access to  `BlockDataManager`, but not the `FileSystemMasterClient`. We need clean this up in another PR.